src/core/server/http/router/request.ts

```diff
@@ -331,3 +345,7 @@ function isRequest(request: any): request is Request {
 export function isRealRequest(request: unknown): request is KibanaRequest | Request {
   return isKibanaRequest(request) || isRequest(request);
 }
+
+function isCompleted(request: Request) {
+  return request.raw.res.writableFinished;
+}
```

Comment:
This is the ~~workaround~~ recommended way to distinguish if a response was properly completed or aborted when a `close` event is fired, given https://github.com/nodejs/node/issues/40775#issuecomment-966085887